.travis.yml

```diff
@@ -1,10 +1,31 @@
 sudo: false
 language: ruby
 rvm:
-- 2.1.2
+  - 2.1.2
+
+gemfile:
+  - chef-client-version.gemfile
+
+env:
+  - CHEF_VERSION=master
+  - CHEF_VERSION=12.4.1
+  # - CHEF_VERSION=12.4.0    # we know 12.4.0 doesn't work and specifically exclude it.
+  - CHEF_VERSION=12.3.0
+  - CHEF_VERSION=12.2.1
+  - CHEF_VERSION=12.1.2
+  - CHEF_VERSION=12.1.1
+  - CHEF_VERSION=12.1.0
+
+matrix:
+  # fast_finish: true
+  allow_failures:
+    - env: CHEF_VERSION=master
+
 script:
+  - bundle install --jobs=3 --retry=3
```

Comment:
I just copied Travis's default `bundle install` command, which (in seeming contravention of the documentation) it was failing to run against a custom gemfile. (I don't actually think we need a custom gemfile, I can just put everything in `Gemfile`, but no need to do that now.)